### PR TITLE
Deployment

### PR DESCRIPTION
**Pull request by macat**

### CODE REVIEW

File: deployment/deployment.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+func init() {
+
+	scheme := runtime.NewScheme()
+	v1.AddToScheme(scheme)
+	factory := serializer.NewCodecFactory(scheme)
+	deserializer = factory.UniversalDeserializer()
+
+	kcfg := &restclient.Config{
+		Host:     "http://localhost:8080",
```
Comment (cored) [inline deployment/deployment.go]: @macat Shouldn't this be an environment variable?

File: deployment/deployment.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+type Step interface {
+	Deploy() error
+}
+
+// DefaultStep implements a deployment step
+type DefaultStep struct {
+	object runtime.Object
+}
+
+var _ Step = &DefaultStep{}
```
Comment (therealplato) [inline deployment/deployment.go]: Why do you create and dump the DefaultStep?

File: deployment/deployment.go (hunk 1 above)
Comment (macat) [inline deployment/deployment.go]: OOo yes, it will be. This is super WIP.

File: deployment/deployment.go (hunk 2 above)
Comment (macat) [inline deployment/deployment.go]: It's just declaring that the defaultstep implements the Step interface.

File: deployment/deployment.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	}
+	return s, nil
+}
+
+// Deploy executes the deployment of a step
+func (s *DefaultStep) Deploy() error {
+	oGVK := s.object.GetObjectKind().GroupVersionKind()
+
+	if oGVK.Kind == "ReplicationController" {
+		rc, err := client.ReplicationControllers("default").Create(object.(*v1.ReplicationController))
```
Comment (therealplato) [inline deployment/deployment.go]: is "default" a kubernetes namespace?

Comment (macat) [inline deployment/deployment.go]: It'll be a parameter in the future.

Comment (macat) [inline deployment/deployment.go]: Yes.

File: deployment/deployment.go (hunk 2 above)
Comment (therealplato) [inline deployment/deployment.go]: Doesn't that just... happen? Since you declare `func (s *DefaultStep) Deploy() error {}`?

Comment (macat) [inline deployment/deployment.go]: Yes, but this way it's more explicit.

Comment (cored) [inline deployment/deployment.go]: @therealplato He is trying to get the compiler to help him. @macat The thing is that I don't think we need that trick if you are doing TDD the compiler will tell you that you are expecting a certain type using an interface and at that moment you will be force to implement those methods.

File: deployment/deployment.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+type Deployment struct {
+	Playbook  playbook.Playbook
+	Variables map[string]string
+	Manifests map[string]*manifest.Manifest
+}
+
+// Deploy executes the deployment
+func (d *Deployment) Deploy() error {
+	for _, task := range d.Playbook.Tasks {
+		if len(task.Manifests) > 0 {
```
Comment (bobbytables) [inline deployment/deployment.go]: Why bother with the length check here?

Comment (macat) [inline deployment/deployment.go]: Because that'll identify what kind of task it is.

Comment (bobbytables) [inline deployment/deployment.go]: We should probably add a method to the `Task` object that can identify that right?

Comment (macat) [inline deployment/deployment.go]: yes, that's a good idea

Comment (bobbytables) [inline deployment/deployment.go]: I'd propose doing a switch statement in here and calling an unexported function on the Deployment instance for whatever needs to happen.

Comment (cored) [inline deployment/deployment.go]: What is the relationship between a task and a manifest I really can't see
it.

**Pull request merged**